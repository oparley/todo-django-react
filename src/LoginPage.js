import React, {Component} from 'react';
import axios from 'axios';

class LoginPage extends Component{
    render(){
        return(
        <div class="modal is-active">
            <div class="box">
                <div className="level"> Login </div>
                <div class="level">
                    <p class="control">
                        <input class="input" type="text" placeholder="username"/>
                    </p>
                </div>
                <div className="level">
                    <p class="control">
                        <input class="input" type="password" placeholder="password"/>
                    </p>
                </div>
                <div className="field is-grouped is-grouped-centered">
                    <p className="control">
                        <a onClick={''} className="button is-success">
                            <span>Login</span>
                        </a>
                    </p>
                </div>
                <div className="field is-grouped is-grouped-centered">
                    <p className="control">
                        <a onClick={''} className="button is-text">
                            <span>Forgot my password</span>
                        </a>
                    </p>
                    <p className="control">
                        <a onClick={''} className="button is-text">
                            <span>Signup</span>
                        </a>
                    </p>
                </div>
            </div>
        </div>
        );
    }
}

export default LoginPage
