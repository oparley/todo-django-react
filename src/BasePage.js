import React, { Component } from 'react'
import axios from 'axios';
import { BrowserRouter as Router, Link, Route, Switch, Redirect } from 'react-router-dom';

import LoginPage from './LoginPage'
import SignupPage from './SignupPage'
import TaskDetail from './TaskDetail'
import HomePage from './HomePage'
import Reports from './Reports'


import { isAuthenticated } from './helpers';



class BasePage extends Component{
    constructor(props){
        super(props)
        this.state = {
            authenticated: isAuthenticated(),
        }
    }

    logout(){
        localStorage.removeItem('token')
        this.setState({authenticated: false})
    }

    login(){
        this.setState({authenticated: true})
    }

    render(){
        let navBar = <nav className="navbar is-link" aria-label="main navigation">
            <div className="navbar-brand">
                <span className="navbar-item">
                    <Link className="button is-link" to="/">TO-DO LIST</Link>
                </span>
            </div>

            <div className="navbar-end">
                <span className="navbar-item">
                    <Link className="button is-link" to="/reports">Reports</Link>
                </span>
                <div className="navbar-item">
                    <a onClick={() => this.logout() }className="button is-link"> Log out </a>
                </div>
            </div>
        </nav>

        let mainContent = <div>
            {navBar}

            <br/>
            <Switch>
                <Route exact path="/" render={() => <HomePage/>}/>
                <Route exact path="/lists/:listid/tasks/:id" render={(e) =>
                    <TaskDetail listid={e.match.params.listid} id={e.match.params.id}/>}
                    />
                <Route exact path="/reports" render={() => <Reports />} />
                <Redirect to="/"/>
            </Switch>
        </div>

        if(!this.state.authenticated){
            mainContent = <Switch>
                <Route exact path="/login" render={() =>
                    <LoginPage authenticated={this.state.authenticated} login={() => this.login()} /> }
                />
                <Route exact path="/signup" render={() =>
                    <SignupPage /> }
                />
                {/* <Route exact path="/recover" render={() =>
                    <LoginPage authenticated={this.state.authenticated} login={() => this.setState({authenticated: true})} /> }
                /> */}
                <Redirect to="/login"/>
            </Switch>
        }

        return(
        <Router>
            {mainContent}
        </Router>

    );}
}

export let API = () => {
    let token = localStorage.getItem('token')
    if(token){
        return axios.create({
            headers: {
                Authorization: `Token ${token}`
            },
        })
    }
    return axios
};

export default BasePage;
