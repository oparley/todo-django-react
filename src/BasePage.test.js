import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import BasePage from './BasePage';
import LoginPage from './LoginPage';

describe('Renders base page', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<BasePage/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('shows login page when token is not set', () => {
  //   const basePage = shallow(<BasePage/>)
  //   expect(basePage.find(LoginPage)).to.have.length(1)
  // });
})
