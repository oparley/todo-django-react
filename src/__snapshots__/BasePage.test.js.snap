// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders base page renders correctly 1`] = `
<div
  className="modal is-active notification is-link"
>
  <div
    className="box"
  >
    <div
      className="level"
    >
       Login 
    </div>
    <div
      className="level"
    >
      <p
        className="control"
      >
        <input
          className="input"
          name="username"
          onChange={[Function]}
          placeholder="username"
          type="text"
          value={undefined}
        />
      </p>
    </div>
    <div
      className="level"
    >
      <p
        className="control"
      >
        <input
          className="input"
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
        />
      </p>
    </div>
    <div
      className="field is-grouped is-grouped-centered"
    >
      <p
        className="control"
      >
        <a
          className="button is-success"
          onClick={[Function]}
        >
          <span>
            Login
          </span>
        </a>
      </p>
    </div>
    <div
      className="field is-grouped is-grouped-centered"
    >
      <p
        className="control"
      >
        <a
          className="button is-text"
        >
          <span>
            Forgot my password
          </span>
        </a>
      </p>
      <p
        className="control"
      >
        <a
          className="button is-text"
          href="/signup"
          onClick={[Function]}
        >
          <span>
            Signup
          </span>
        </a>
      </p>
    </div>
  </div>
</div>
`;
